### DISCUSSION TITLE
On Android, session is not expired like iOS

### DISCUSSION THREAD

**Discussion by chnouman** [ThexXTURBOXx/flutter_web_auth_2#115] [Q&A]
On iOS, when a user logs in and then uninstalls the app, the browser does not retain the session history. However, on Android, the browser does persist the history even if the app is reinstalled. I attempted to control this behavior using intentFlags on android, but it didn't seem to work. Could anyone provide guidance or a solution to prevent the browser from retaining history on Android? Your help would be greatly appreciated.

**Comment by ThexXTURBOXx** [comment]
Good question. This is probably something you are better off asking in the Google/Android/AndroidX forums.
`flutter_web_auth_2` uses `androidx.browser` internally to create the custom tabs. If you delete the data of Google Chrome (or whatever custom tabs provider you are using), you will probably be prompted to log in again.
Otherwise, I can just refer to comments such as [this](https://github.com/ThexXTURBOXx/flutter_web_auth_2/issues/114#issuecomment-2230296692)

**Comment by chnouman** [comment]
Thank you for your input. I have a couple of follow-up questions:

Determining the Browser Used for Authentication: I’m currently using the in_app_webview library and trying to call signout URL for clearing the cache and specifying `com.android.chrome` as the package name for custom tabs. However, since not all devices have Chrome installed, I need a reliable way to determine which browser was used for authentication and handle cases where Chrome or other specific browsers might not be available. Could you suggest how to handle this, or if there’s a way to programmatically detect the browser used?

I appreciate any further insights or suggestions you might have on this topic.

Thank you for your assistance!

**Reply by ThexXTURBOXx** [reply]
I am not really aware of a "nice" way of achieving this, but a possible way would be to check for the presence of your specified browsers in the same way we do it [here](https://github.com/ThexXTURBOXx/flutter_web_auth_2/blob/master/flutter_web_auth_2/android/src/main/kotlin/com/linusu/flutter_web_auth_2/FlutterWebAuth2Plugin.kt#L83-L148).

**Reply by chnouman** [reply]
Hmm ok, I'll try to do so. Btw thanks for your input this was my plan if nothing works.
